torch/csrc/distributed/rpc/rref_impl.cpp

```diff
@@ -105,6 +105,15 @@ const ForkId& UserRRef::forkId() const {
 }
 
 IValue UserRRef::toHere() {
+  // see Note [Best-Effort Check on Deleted UserRRefs]
+  TORCH_CHECK(
+      !deletedOnOwner_,
+      "User RRef with RRefId=",
+      rrefId(),
+      " and ForkId=",
+      forkId(),
+      " has been deleted. Cannot call to_here() on it after deletion.");
```

Comment:
This should be a robust check? since after entering shutdown(gracefuk) and run past _wait_all_workers(), there should be no to_here() calls.

more to_here() calls could happen after shutdown(), but the deletedOnOwner_ status should be correct.

Reply:
I agree, if we assume application code is single-threaded.